src/curves/g1_curve.rs

```diff
@@ -1,23 +1,27 @@
+//! Elliptic curve `y^2=x^3+3`` over the prime field GF(101).
+
 use super::*;
 
+// TODO: An empty struct like this and G2Curve is a code smell. We should probably
```

Comment:
This was a design @thor314 had taken I believe from artworks? (correct me if I am wrong). I am okay thinking about a single struct too.

Reply:
You could at least do a single struct like 
```rust
CurveGroup<F: FiniteField> {
    _phantom: F,
}
```
This way you can just use one struct and define it over whatever field you'd like.

I think there's probably something even nicer we could do to better utilize the `Curve` as we have as of now.